packages/cli-kit/src/ui.ts

```diff
@@ -167,23 +167,33 @@ export async function terminateBlockingPortProcessPrompt(port: number, stepDescr
   return choice.value === 'finish'
 }
 
-export const keypress = async () => {
+export const keypress = async (key?: string) => {
   return new Promise((resolve, reject) => {
-    const handler = (buffer: Buffer) => {
-      process.stdin.setRawMode(false)
-      process.stdin.pause()
-
-      const bytes = Array.from(buffer)
-
-      if (bytes.length && bytes[0] === 3) {
+    const handler = (data: string) => {
+      if (data === '\u0003') {
+        process.stdin.pause()
+        process.stdin.setRawMode(false)
         debug('Canceled keypress, User pressed CTRL+C')
         reject(new AbortSilentError())
+        return
       }
-      process.nextTick(resolve)
+
+      if (key) {
+        if (data === key) {
+          process.nextTick(resolve)
+        } else {
+          return
+        }
+      } else {
+        process.nextTick(resolve)
+      }
+
+      process.stdin.pause()
+      process.stdin.setRawMode(false)
     }
 
     process.stdin.resume()
     process.stdin.setRawMode(true)
-    process.stdin.once('data', handler)
+    process.stdin.on('data', handler)
```

Comment:
So are we going to keep listening for keypresses here? why the change?

Reply:
Could you check if this change affects the authentication flow?

Reply:
I believe calling `pause()` should detach this, but I'm going to double check.